scripts/dev/upstream_merge/upstream_merge_and_test.py

```diff
@@ -0,0 +1,333 @@
+
+    return (0,None)
+
+def create_merge_branch(git_obj,merge_branch_name):
+    if git_obj.branch_exists(merge_branch_name):
+        git_obj.checkout_branch(git_obj.local_base_branch)
+        git_obj.delete_branch(merge_branch_name)
+
+    if git_obj.checkout_branch(merge_branch_name,create=True)[0] != 0:
+        print(f"\n    Error creating {merge_branch_name}. Exiting")
+        return (1,f"\n    Error creating {merge_branch_name}. Exiting")
+    
+    return (0,None)
+
+def prepare_for_merge(git_obj, merge_branch_name, force_checkout):
+    print(f"{git_obj.local_repo}")
+
+    base_branch_details = switch_to_base_branch_and_pull(git_obj,force_checkout)
+    if base_branch_details[0] != 0:
+        return base_branch_details
+    
+    fetch_details = fetch_upstream(git_obj)
+    
+    if fetch_details[0] != 0:
+        return fetch_details
+    
+    repo_details = create_merge_branch(git_obj, merge_branch_name)
+
+    if repo_details[0] != 0:
+        return repo_details
+    
+    return (0,None)
+
+def merge_upstream(git_obj,force_checkout, merge_branch_name, skip_merge):
+    if skip_merge:
+        return (0," Has Been Skipped")
+    
+    merge_prepare_details = prepare_for_merge(git_obj,merge_branch_name,force_checkout)
+    
+    if merge_prepare_details[0] != 0:
+        return merge_prepare_details
+    
+    commit_before_merge = git_obj.get_current_commit()
+    
+    merge_result = git_obj.merge_branch(f"{git_obj.upstream_repo_name}/{git_obj.upstream_branch}", "Merge latest upstream")
+
+    if merge_result[0] == 0:        
+        diff_output=git_obj.diff()    
+        if (git_obj.get_current_commit() == commit_before_merge) or diff_output == (0,''):
+            return (0,None)
+        else:
+            return (0,diff_output[1])
+    else:
+        return (1,merge_result[1])
+
+def push_branch_and_create_PR(git_obj, merge_branch_name, work_item_id, username):
+    push_details = push_branch(git_obj,merge_branch_name)
+    if push_details[0] != 0:
+        return push_details
+    
+    # PR_details = create_PR(git_obj,merge_branch_name,work_item_id,username)
+    # if PR_details[0] !=0:
+    #     return PR_details
+
+    return (0,None)
+
+def push_branch(git_obj,merge_branch_name):
+    if git_obj.add_remote(git_obj.fork_name, git_obj.fork_url)[0] != 0:
+        print(f"\n    Error adding remote repository {git_obj.fork_name} using {git_obj.fork_url}. Exiting")
+        return (1,f"\n    Error adding remote repository {git_obj.fork_name} using {git_obj.fork_url}. Exiting")
+    
+    if git_obj.branch_exists(merge_branch_name,git_obj.fork_name,check_on_remote=True):
+        if git_obj.push(merge_branch_name, git_obj.fork_name, delete=True)[0] != 0:
+            print(f"\n    Failed to delete branch {merge_branch_name} on {git_obj.fork_name}")
+            return (1, f"\n    Failed to delete branch {merge_branch_name} on {git_obj.fork_name}")
```

Comment:
This error message makes more sense when you say, "Failed to delete the remote branch and push the local branch {merge_branch_name} on {git_obj.fork_name}", right?

If there are different error codes for delete failure vs merge failure here, it would be helpful if you pass that back to the calling function, instead of just passing a 1 for any error that might occur.